firmware/components/driver_sensor_am2320/driver_am2320.c

```diff
@@ -0,0 +1,155 @@
+/*
+ * Driver for the AM2320 I2C temperature and humidity sensor
+ * Jeroen Bos (Hieronymox)
+ *
+ */
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_log.h"
+#include "esp_timer.h"
+#include <sdkconfig.h>
+#include <driver/gpio.h>
+
+#include <include/driver_i2c.h>
+#include "include/driver_am2320.h"
+
+static const char *TAG              = "driver_am2320";
+static bool driver_am2320_init_done = false;
+
+#ifdef CONFIG_DRIVER_AM2320_ENABLE
+
+static float cached_temperature = 0;
+static float cached_humidity    = 0;
+static uint64_t cache_time      = 0;
+
+esp_err_t driver_am2320_init(void)
+{
+    if (driver_am2320_init_done)
+        return ESP_OK;
+
+    ESP_LOGD(TAG, "init called");
+
+    if (CONFIG_I2C_MASTER_FREQ_HZ > 100000) {
+        ESP_LOGE(
```

Comment:
Maybe replace this with an error during compile time

#if CONFIG_I2C_MASTER_FREQ_HZ > 100000
#error "I2C interface speed is set to more than 100kHz, the AM2320 sensor supports a speed of at most 100kHz."
#endif

Reply:
Removed the if and added error macro to header file